src/screens/WhotoFollow/index.js

```diff
@@ -177,65 +177,70 @@ const WhotoFollow = () => {
-            setProfileId({
-              anonProfileId: anonymousUserId,
-              signedProfileId: userId,
-              token: res.token,
-              anonymousToken: res.anonymousToken
+      registerUser(data)
+        .then(async (res) => {
+          setFetchRegister(false);
+          if (res.code === 200) {
+            TokenStorage.set(res);
+            setToken(res.token);
+            try {
+              const userId = await JwtDecode(res.token).user_id;
+              const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
+              setProfileId({
+                anonProfileId: anonymousUserId,
+                signedProfileId: userId,
+                token: res.token,
+                anonymousToken: res.anonymousToken
+              });
+            } catch (e) {
+              crashlytics().recordError(new Error(e));
+            }
+            showMessage({
+              message: 'Welcome to Better Social',
+              type: 'success',
+              backgroundColor: colors.holytosca
+            });
+            setTimeout(() => {
+              create();
+              setImage(null, usersDispatch);
+              setInitialValue({id: res.token});
+            }, 2000);
+          } else {
+            crashlytics().recordError(new Error(res));
+            showMessage({
+              message: 'Cannot connect to server, please try again later',
+              type: 'danger',
+              backgroundColor: colors.red
             });
-          } catch (e) {
-            crashlytics().recordError(new Error(e));
           }
-          showMessage({
-            message: 'Welcome to Better Social',
-            type: 'success',
-            backgroundColor: colors.holytosca
-          });
-          setTimeout(() => {
-            create();
-            setImage(null, usersDispatch);
-            setInitialValue({id: res.token});
-          }, 2000);
-        } else {
-          crashlytics().recordError(new Error(res));
+        })
+        .catch((error) => {
+          crashlytics().recordError(new Error(error.response));
+          setFetchRegister(false);
           showMessage({
             message: 'Cannot connect to server, please try again later',
             type: 'danger',
             backgroundColor: colors.red
           });
-        }
-      })
-      .catch((error) => {
-        crashlytics().recordError(new Error(error.response));
-        setFetchRegister(false);
-        showMessage({
-          message: 'Cannot connect to server, please try again later',
-          type: 'danger',
-          backgroundColor: colors.red
         });
+    } catch (e) {
+      crashlytics().recordError(new Error(e));
+      setFetchRegister(false);
+      showMessage({
+        message: 'Cannot connect to server, please try again later',
+        type: 'danger',
+        backgroundColor: colors.red
       });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this code could be improved. Currently, if an error occurs during the image upload process (lines 183-187), it is caught and logged, but the execution continues. This might lead to unexpected behavior as `data.users.profile_pic_path` may not be updated with the correct URL of the uploaded image. It would be better to handle this error properly, either by stopping the execution or by providing a default value for `data.users.profile_pic_path`.

```diff
    try {
      const profilePic = usersState?.photoUrl;
      if (profilePic && profilePic !== DEFAULT_PROFILE_PIC_PATH) {
         const uploadedImageUrl = await ImageUtils.uploadImageWithoutAuth(
           data?.users?.profile_pic_path
         );
         data.users.profile_pic_path = uploadedImageUrl?.data?.url;
+         } else {
+           // Provide a default value or throw an error
+           data.users.profile_pic_path = DEFAULT_IMAGE_URL;
       }
 
      registerUser(data)
...
```

Also, there are multiple places where the same error message is shown to the user (lines 219-223, 229-233, and 238-242). To avoid repetition and improve maintainability, you can define this message once and reuse it.

```diff
+    const errorMessage = {
+      message: 'Cannot connect to server, please try again later',
+      type: 'danger',
+      backgroundColor: colors.red
+    };

...

            crashlytics().recordError(new Error(res));
-            showMessage({
-              message: 'Cannot connect to server, please try again later',
-              type: 'danger',
-              backgroundColor: colors.red
-             });
+            showMessage(errorMessage);

...

           showMessage({
-              message: 'Cannot connect to server, please try again later',
-              type: 'danger',
-              backgroundColor: colors.red
-            });

+            showMessage(errorMessage);

...

      showMessage({
-         message: 'Cannot connect to server, please try again later',
-         type: 'danger',
-         backgroundColor: colors.red
-        });

+       showMessage(errorMessage);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->